### PR TITLE
Make REPORT_STEP optional in GEN_DATA and GENERAL_OBSERVATION

### PR DESCRIPTION
**Pull request by oyvindeide**
**Issue**
Resolves #my_issue

**Approach**
_Short description of the approach_

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Updated documentation
- [x] Ensured new behaviour is tested

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: src/ert/config/gen_data_config.py [hunk 1]
```diff
@@ -15,36 +15,39 @@ class GenDataConfig(ResponseConfig):
...
+        _run_path = Path(run_path)
+        if self.report_steps:
+            for report_step in self.report_steps:
+                filename = filename_fmt % report_step
+                if not Path.exists(_run_path / filename):
+                    errors.append(f"{self.name} report step {report_step} missing")
+                    continue
+                datasets.append(_read_file(filename, report_step))
+        else:
+            datasets.append(_read_file(filename_fmt, 0))
```
Comment (sondreso) [inline src/ert/config/gen_data_config.py:50]: We are not guarding against the file not existing here, think we should provide a reasonable error message in this scenario.

File: tests/unit_tests/c_wrappers/res/enkf/test_load_forward_model.py [hunk 2]
```diff
@@ -246,3 +246,31 @@ def test_that_all_decativated_values_are_loaded(prior_ensemble):
...
+    with open(run_path / "response.out", "w", encoding="utf-8") as fout:
+        fout.write("\n".join(["1", "2", "3"]))
+    with open(run_path / "response.out_active", "w", encoding="utf-8") as fout:
+        fout.write("\n".join(["1", "0", "1"]))
+
+    facade = LibresFacade(ert)
+    facade.load_from_forward_model(prior_ensemble, [True], 0)
+    assert list(
+        facade.load_gen_data(prior_ensemble, "RESPONSE", 0).dropna().values.flatten()
+    ) == [1.0, 3.0]
```
Comment (sondreso) [inline tests/unit_tests/c_wrappers/res/enkf/test_load_forward_model.py:276]: Missing newline :slightly_smiling_face:

File: tests/unit_tests/c_wrappers/res/enkf/test_load_forward_model.py [hunk 3]
```diff
@@ -246,3 +246,31 @@ def test_that_all_decativated_values_are_loaded(prior_ensemble):
     assert (
         len(facade.load_gen_data(prior_ensemble, "RESPONSE", 0).values.flatten()) == 1
     )
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+def test_load_forward_model_gen_data_no_percent(prior_ensemble):
```
Comment (sondreso) [inline tests/unit_tests/c_wrappers/res/enkf/test_load_forward_model.py:252]: test that?

File: docs/getting_started/configuration/poly_new/guide.rst [hunk 4]
```diff
@@ -305,12 +305,12 @@ To enable ERT to read the simulation results, you'll need to use the :ref:`GEN_D
 
 .. code-block:: shell
 
-    GEN_DATA POLY_RES RESULT_FILE:poly_%d.out REPORT_STEPS:0 INPUT_FORMAT:ASCII
+    GEN_DATA POLY_RES RESULT_FILE:poly.out REPORT_STEPS:0 INPUT_FORMAT:ASCII
```
Comment (dafeda) [inline docs/getting_started/configuration/poly_new/guide.rst:308]: Can we get rid of `REPORT STEPS` here?

Comment (oyvindeide) [reply docs/getting_started/configuration/poly_new/guide.rst:308]: Yes, I got lost in rebasing, will fix it

File: docs/getting_started/configuration/poly_new/guide.rst [hunk 5]
```diff
@@ -305,12 +305,12 @@ To enable ERT to read the simulation results, you'll need to use the :ref:`GEN_D
...
+    GEN_DATA POLY_RES RESULT_FILE:poly.out
 
 2. **Understanding the Arguments**:
 
 - **POLY_RES**: Name of this result set.
-- **RESULT_FILE:poly_%d.out**: Path to the file with the simulation results. 
+- **RESULT_FILE:poly.out**: Path to the file with the simulation results.
   The ``%d`` is always ``0``, but must be specified. (a requirement from earlier ERT versions).
 - **``REPORT_STEPS:0``**: Generally set to 0.
 - **``INPUT_FORMAT:ASCII``**: Specifies that the file is a text file (`ASCII` stands for "American Standard Code for Information Interchange").
```
Comment (sondreso) [inline docs/getting_started/configuration/poly_new/guide.rst:316]: These do not make sense anymore, but they should be included in the general documentation for the keyword if they are not at the moment.
```suggestion
```

**Review by sondreso — APPROVED** [review]
LGTM, only small comment about docs!

**Pull request merged**